discovery_engine_core/providers/src/newscatcher.rs

```diff
@@ -49,31 +49,37 @@ pub struct Article {
     pub id: String,
 
     /// The title of the article.
+    #[serde(deserialize_with = "deserialize_null_default")]
     pub title: String,
 
     /// How well the article is matching your search criteria.
     #[serde(rename(deserialize = "_score"))]
     pub score: Option<f32>,
 
     /// The page rank of the source website.
+    #[serde(deserialize_with = "deserialize_null_default")]
```

Comment:
Since when we rank without cois we use rank for the headlines I think here would be better to have `usize::MAX` instead of 0 to penalize documents without infos.